dialogflow/spec/session_entity_type_spec.rb

```diff
@@ -0,0 +1,79 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "rspec"
+require "google/cloud/dialogflow"
+
+require_relative "../session_entity_type_management"
+require_relative "../detect_intent_texts"
+require_relative "../entity_type_management"
+
+describe "Session_entity_type Management" do
+
+  before do
+    @project_id = ENV["GOOGLE_CLOUD_PROJECT"]
+    @session_id = "fake_session_for_testing"
+    @entity_type_display_name = 'fake_display_name_for_testing'
+    @entity_values = ['fake_entity_value_1', 'fake_entity_value_2']
+  end
+
+  example "create session_entity_type" do
+    # create an entity type to be overridden
+    create_entity_type project_id: @project_id,
+                       display_name: @entity_type_display_name,
+                       kind: :KIND_MAP
+
+    # create a session
+    detect_intent_texts project_id:    @project_id,
+                        session_id:    @session_id,
+                        texts:         ['hi'],
+                        language_code: 'en-US' 
+
+    create_session_entity_type project_id: @project_id,
+                               session_id: @session_id,
+                               entity_type_display_name: @entity_type_display_name,
+                               entity_values: @entity_values
+
+    expectation = expect {
+      list_session_entity_types project_id: @project_id, session_id: @session_id
+    }
+
+    expectation.to output(/#{@session_id}/).to_stdout
+    expectation.to output(/#{@entity_type_display_name}/).to_stdout
+    expectation.to output(/#{@entity_values[0]}/).to_stdout
+    expectation.to output(/#{@entity_values[1]}/).to_stdout
+  end
+
+  example "delete session_entity_type" do
+    delete_session_entity_type project_id: @project_id,
```

Comment:
Are you deleting something that doesn't exist? How can you call `delete` before creating something to delete?

For a DELETE test, I would expect it to flow as following:

```python
ARRANGE
  ->create thing
  assert thing
ACT
   ->delete thing
ASSERT
  assert NOT thing
```

Reply:
Similarly, for a CREATE test, I would expect...

```python
ARRANGE
  assert NOT thing
ACT
   ->create thing
ASSERT
  assert thing
```

Reply:
risking making these tests not strictly unit tests, since we need to test both creation and deletion and the CI runs the tests in a file in sequence (is this true?) I had an earlier test doing creation, and a later test doing deletion, testing both functionalities in a suite of two calls instead of 4.